### ISSUE TITLE
[docs] Search button wandering off

### ISSUE THREAD

**Issue by qw-in** [supabase/supabase#7381]
# Bug report

## Describe the bug

![Screenshot_2022-06-22-12-03-34-51_3aea4af51f236e4932235fdada7d1643.jpg](https://user-images.githubusercontent.com/19194187/175116757-3fae7f96-83a1-4991-b939-303c0dc8bc98.jpg)

## Expected behavior

Search button is contained by header

## System information

- OS: Android
- Browser (if applies) Firefox

**Comment by jaysoni-ash42** [comment]
@Isaiah-Hamilton  can i work on this task

**Comment by Isaiah-Hamilton** [comment]
@jaysoni-ash42 all yours

**Comment by jaysoni-ash42** [comment]
@Isaiah-Hamilton  weird thing locally the search icon is inline even after the builded web but on hosted it is inlined 
<img width="706" alt="Screen Shot 2022-06-23 at 12 23 13" src="https://user-images.githubusercontent.com/68722497/175232274-0e4edbd9-1ecb-46ce-a0ad-5ec83630cd71.png">

**Comment by Isaiah-Hamilton** [comment]
@jaysoni-ash42 that is pretty weird. do you have any ideas what might be happening here 🤔?

**Comment by justinduplain** [comment]
@jaysoni-ash42, @Isaiah-Hamilton,

In dev tools: adding the `top: var(--ifm-navbar-padding-vertical);` line as per below snippet fixes the issue in the live site.
 
I searched for these CSS settings in the main repo and found nothing. 

I think the disconnect might be happening in the imported supabase/ui  `IconSearch` component and that may be why the dev version differs from the live site? Can anyone confirm that reasoning? 


```
@media (max-width: 996px)
.searchBox_UMag, .searchBox_ZlJk {
    position: absolute;
    right: var(--ifm-navbar-padding-horizontal);
    top: var(--ifm-navbar-padding-vertical);
}
```

**Comment by jaysoni-ash42** [comment]
> IconSearch

@Isaiah-Hamilton  @justinduplain  yup the inlinement is because of the difference of version on the deployment and the dev version.

**Comment by qw-in** [comment]
`searchBox_node_modules-docusaurus-theme-search-typesense-src-theme-SearchBar-styles-module`

What a class name!

**Comment by burmecia** [comment]
That class name is generated from the Typesense component https://github.com/typesense/docusaurus-theme-search-typesense, so its name looks ridiculous. And the css name is shorten when it is built on prod but not on local dev, so the customised css styles in `web/src/css/custom.css` won't apply to prod, that's why it cannot been reproduced on local dev.

To fix this issue, I think we only need to override the `position` property to the `searchBox` containers like below:

```
div[class^="searchBox_"] {
  position: inherit !important;
}
```

and then we can use it to replace all the customised css styles from the doc search in `web/src/css/custom.css`:

```
/*
* typesense doc search input
*/
.searchBox_node_modules-\@docusaurus-theme-classic-lib-theme-Navbar-Search-styles-module {
  padding: 0 !important;
}

.searchBox_node_modules-docusaurus-theme-search-typesense-src-theme-SearchBar-styles-module {
  position: relative !important;
  right: 0 !important;
  padding: 0 !important;
  margin-left: 16px;
}

.searchBox_node_modules-\@docusaurus-theme-classic-lib-theme-Navbar-Search-styles-module button {
  padding: 8px 14px !important;
  font-size: inherit !important;
}

.searchBox_node_modules-\@docusaurus-theme-classic-lib-theme-Navbar-Search-styles-module svg {
  width: 16px;
  height: 16px;
}
```

**Comment by MildTomato** [comment]
ah i didn't realize the classnames changed in prod 🙈 my bad

**Comment by MildTomato** [comment]
feel free to apply dirty fixes on this, since we are replacing this website soon anyway

**Comment by burmecia** [comment]
This issue has been fixed by #7609 .
